armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -184,6 +184,26 @@ def construct(
                         # learn mult from grid definition
                         c.setDimension("mult", len(c.spatialLocator))
 
+                if componentDesign.latticeIDs:
+                    # the user has given this component latticeIDs, implying that it
+                    # should be present in the blocks associated grid. check that
+                    # it is in the grid the correct number of times, otherwise
+                    # their blueprints are probably wrong
+                    idsInGrid = list(gridDesign.gridContents.values())
+                    totalInstancesOfComponentInGrid = 0
+                    for latticeID in componentDesign.latticeIDs:
+                        nInstances = len([i for i in idsInGrid if i == str(latticeID)])
+                        totalInstancesOfComponentInGrid += nInstances
+
+                    expectedNumberOfInstances = c.getDimension("mult")
+                    if totalInstancesOfComponentInGrid != expectedNumberOfInstances:
+                        raise ValueError(
+                            f"Component {c} in block blueprint '{self.name}' is expected to be present in the "
+                            f"associated block grid {expectedNumberOfInstances} times, but it "
+                            f"is found {totalInstancesOfComponentInGrid} times. "
+                            "Check that the component's latticeIDs align with the block's grid."
+                        )
```

Comment:
```suggestion
                    numInstances = 0
                    for latticeID in componentDesign.latticeIDs:
                        nInstances = len([i for i in idsInGrid if i == str(latticeID)])
                        numInstances += nInstances

                    mult = c.getDimension("mult")
                    if numInstances != mult:
                        raise ValueError(
                            f"Component {c} in block blueprint '{self.name}' is expected to be present in the "
                            f"associated block grid {mult} times, but it is found {numInstances} times. "
                            "Check that the component's latticeIDs align with the block's grid."
                        )
```

Reply:
I won't die on this hill. Just turn it down if you don't like it.

Long variable names SHOULD, in theory, be more clear. But they were so long I had problems reading the code (particularly that error message).

Up to you.